From: Stephane Glondu <steph@glondu.net>
Date: Sat, 19 Jun 2010 20:03:48 -0700
Subject: [PATCH] Fix build with OCaml 3.12.0

The quotation <:ctyp< $a$ $b$ >> doesn't work any more with OCaml
3.12.0+beta1; we manually expand it.

Bug-OCaml: http://caml.inria.fr/mantis/view.php?id=5080
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=586493
Signed-off-by: Stephane Glondu <steph@glondu.net>
---
 src/syntax/pa_strings/pa_format.ml |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/syntax/pa_strings/pa_format.ml b/src/syntax/pa_strings/pa_format.ml
index 0b40904..aa402ca 100644
--- a/src/syntax/pa_strings/pa_format.ml
+++ b/src/syntax/pa_strings/pa_format.ml
@@ -246,7 +246,7 @@ struct
         <:expr< fun paren out x -> BatIO.nwrite out "<abstract>" >>
     | <:ctyp@_loc< $id:id$ >> ->
         <:expr< $id:map_id id$ >>
-    | <:ctyp@_loc< $a$ $b$ >> ->
+    | Ast.TyApp (_loc, b, a) ->
         <:expr< $vprinter_of_ctyp b$ $vprinter_of_ctyp a$ >>
     | <:ctyp@_loc< $tup:t$ >> ->
         let l = Ast.list_of_ctyp t [] in
-- 
